spec/rubocop/cop/rspec/sort_metadata_spec.rb

```diff
@@ -4,21 +4,80 @@
   it 'does not register an offense when using only symbol metadata ' \
      'in alphabetical order' do
     expect_no_offenses(<<~RUBY)
-      RSpec.describe 'Something', :a, :b do
+      describe 'Something', :a, :b do
       end
     RUBY
   end
 
   it 'registers an offense when using only symbol metadata, ' \
      'but not in alphabetical order' do
     expect_offense(<<~RUBY)
-      RSpec.describe 'Something', :b, :a do
-                                  ^^^^^^ Sort metadata alphabetically.
+      describe 'Something', :b, :a do
+                            ^^^^^^ Sort metadata alphabetically.
       end
     RUBY
 
     expect_correction(<<~RUBY)
-      RSpec.describe 'Something', :a, :b do
+      describe 'Something', :a, :b do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense for a symbol metadata before a non-symbol ' \
+     'argument' do
+    expect_no_offenses(<<~RUBY)
+      describe 'Something', :z, :a, variable, foo: :bar do
+      end
+    RUBY
+  end
+
+  it 'registers an offense only for trailing symbol metadata not in ' \
+     'alphabetical order' do
+    expect_offense(<<~RUBY)
+      describe 'Something', :z, :a, variable, :c, :b do
+                                              ^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      describe 'Something', :z, :a, variable, :b, :c do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when a symbol metadata not in alphabetical order ' \
+     'is before a variable argument being the last argument ' \
+     'as it could be a hash' do
```

Comment:
This seems a bit strange to me. If `some_hash` “could be a hash”, it could also be e.g. an array. And if we sort symbol metadata before an array variable, why don’t we _always_ sort symbol metadata before a variable?

Reply:
That's a valid point. Indeed each group of symbols could be alphabetically sorted independently.

This PR was originally addressing #1946 and is now approved. If that makes sense, could your point be registered as a separate issue to be addressed later? Feel free to mention me and I can work on a PR for it.

Reply:
My comment was not meant to block this PR; I’m find with merging this as-is.